test/cluster.cpp

```diff
@@ -61,22 +61,26 @@ namespace
 class TempFile
 {
   int fd_;
+#ifndef _WIN32
   std::string path_;
+#endif
 public:
-  TempFile()
+  explicit TempFile(const char* name)
   {
 #ifdef _WIN32
+    const int n = strlen(name);
     wchar_t cbase[MAX_PATH];
     wchar_t ctmp[MAX_PATH];
-    GetTempPathW(MAX_PATH-14, cbase);
+    wchar_t wname[MAX_PATH];
+    mbstowcs(wname, name, n+1);
```

Comment:
I recently discover the c++ wstring_convert class to convert string encoding thanks to @MiguelRocha (openzim/zim-tools:src/zimdump.cpp@master#L64,

This seems a better way (api) to convert the path for windows. Maybe not enough to change all our code, but for new one I prefer to use wstring_convert

Reply:
Done